monitoring: Add monitoring architecture doc

anmolbabu opened the pull request:
Add monitoring architecture doc

Issue: https://github.com/Tendrl/documentation/issues/40
Jira: https://tendrl.atlassian.net/browse/TEN-49, https://github.com/Tendrl/documentation/issues/32, https://tendrl.atlassian.net/browse/TEN-84
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

anmolbabu commented:
@nthomas-redhat @shtripat @brainfunked @anupnivargi @anupnivargi Please review

brainfunked reviewed (changes requested):
In the diagram, I would suggest colour coding the boxes to depict the stacks the boxes belong to. In this scenario core and performance monitoring stacks would need to be depicted with their specific colours. The other boxes could be left colourless.

shtripat commented on tendrl-monitoring-architecture.adoc:
```diff
@@ -0,0 +1,78 @@
+   regarding the Operations being executed.
+** Status of every relevant Tendrl and Storage System process/component that
+   needs to be monitored.
+** Current time series data needed for automated conditional triggers.
```
Is it correct?

shtripat commented on tendrl-monitoring-architecture.adoc:
```diff
@@ -0,0 +1,78 @@
+** Process monitoring for the various system and Storage System daemons,
+   including the Tendrl Bridge via systemd.
+** Invocation of host-local Provisioning Operations.
+** Monitoring and gathering of the system’s status, including collection of
```
node agent does gathering of time series data??

brainfunked replied:
This was an older document, which needs to be updated with the appropriate details from the newly finalised monitoring architecture.

shtripat replied:
@anmolbabu can you please update this?

anmolbabu committed:
Add monitoring architecture doc

Add monitoring architecture doc

Issue: https://github.com/Tendrl/documentation/issues/40
Jira: https://tendrl.atlassian.net/browse/TEN-49, #32, https://tendrl.atlassian.net/browse/TEN-84
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

anmolbabu commented:
@brainfunked @shtripat @nthomas-redhat @anupnivargi I have incorporated comments. Please review.

brainfunked commented on tendrl-monitoring-architecture.adoc:
```diff
@@ -0,0 +1,197 @@
+  ** Alert subscription management for users in tendrl core(This functionality
+     will be achieved for users in tendrl core. These subscriptions will be
+     stored in tendrl’s central store).
+  ** A BSD Licensed python tool by name flask is used for serving the
```
This doesn't need to be part of the design discussion. Perhaps can be explained in a separate section on the selection of tools.

anmolbabu replied:
Done. Removed it from here..

brainfunked commented on tendrl-monitoring-architecture.adoc:
```diff
@@ -0,0 +1,197 @@
+        have it or node-agent/bridge was started before starting performance
+        monitoring application, a flow will be intiated internally to generate
+        collectd configurations once the performance monitoring application is
+        started.
```
The monitoring application should be treated as an add-on stack. This means that the corresponding code, templates, configurations, flows etc. all need to be shipped for the monitoring stack should be shipped separately, with an assumption that the core stack is already started.

I would suggest that the flow framework from the core stack be reused for deploying the monitoring stack.
1. Install the monitoring application.
2. Monitoring application connects to etcd and uploads the default monitoring configuration and flows.
3. Monitoring application triggers the monitoring provisioning flow on every node for node monitoring via the core API.
4. The provisioning flow for the monitoring stack installs the collectd-related pieces on every node and generates the configuration files from templates shipped with the collectd plugins on nodes.

anmolbabu replied:
Done

fbalak commented on api/monitoring-api.adoc:
```diff
@@ -0,0 +1,77 @@
+Sample Request
+
+----------
+GET /moitoring/nodes/<node_name>/monitored_types
```
There is probably typing error: monitoring instead of moitoring.

anmolbabu replied:
Yes, that was a typo :) .. Thanks for pointing..

anmolbabu committed:
Incorporated comments

Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

anmolbabu commented:
@brainfunked @nthomas-redhat @shtripat @r0h4n @fbalak Comments incorporated. Please review
